modules/programs/neovim.nix

```diff
@@ -174,8 +159,14 @@ in {
         '';
       };
 
+      # Currently, upstream Neovim is pinned on Lua 5.1 for LuaJIT support.
+      # This will need to be updated if Neovim ever migrates to a newer
+      # version of Lua.
       extraLuaPackages = mkOption {
-        type = with types; either extraLua51PackageType (listOf package);
+        type = with types;
```

Comment:
rather than silently doing it, I would prefer to warn the user, make it work but with a deprecation notice.